sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java

```diff
@@ -226,7 +226,10 @@ private String generateQueryBody(@NonNull Criteria criteria, @NonNull List<Pair<
     @NonNull
     private Pair<String, List<Pair<String, Object>>> generateQueryBody(@NonNull CosmosQuery query, @NonNull final AtomicInteger counter) {
         final List<Pair<String, Object>> parameters = new ArrayList<>();
-        String queryString = this.generateQueryBody(query.getCriteria(), parameters, counter);
+        String queryString = "";
+        if (query.getCriteria() != null) {
+            queryString = this.generateQueryBody(query.getCriteria(), parameters, counter);
```

Comment:
So if the criteria is null, will the query body be completely empty? how will that work for non-criteria queries?

Reply:
I was confused initially as well - but looks like it is just coming from unintuitive variable names - generateQueryBody does not generate the query text/body but just the where clause

Reply:
Correct I was confused at first also, many of the new tests I added do have criteria as null

Reply:
Awesome, thanks @trande4884